tests/utils/test_requirements_utils.py

```diff
@@ -697,3 +697,24 @@ def predict(self, context, model_input, params=None):
         mock_warning.assert_not_called()
         assert "pandas" in modules
         assert "sklearn" in modules
+
+
+def test_capture_imported_modules_extra_env_vars():
+    class TestModel(mlflow.pyfunc.PythonModel):
+        def predict(self, context, model_input, params=None):
+            assert os.environ["TEST"] == "test"
+            import sklearn  # noqa: F401
```

Comment:
```suggestion
```

do we need this line?

Reply:
Maybe no, just another check to make sure the predict function runs correctly